NuGet/NuGet3
# fix NuGet/Home#622 by correctly handling a locked lockfile

fixes NuGet/Home#622

To test it out:
1. Clone this repo (if you don't already have it)
2. `git checkout anurse/622-lock-dat-file`
3. `.\build`
4. `cd misc\RestoreTests\LockedLockFile`
5. Clear your `%USERPROFILE%\.nuget\packages` directory
6. Run `..\..\..\nuget3 restore --source https://www.myget.org/F/dotnet-core/api/v2`

The expected result is that, **despite** the `*` in project.json and the fact that there are higher versions on the feed, only versions `4.0.x-beta-22927` are restored.

/cc @davidfowl @lodejard @ericstj @emgarten @yishaigalatzer
